pkg/controller/stackconfigpolicy/kibana_config_settings.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package stackconfigpolicy
+
+import (
+	"context"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	kibanav1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/kibana/v1"
+	policyv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/stackconfigpolicy/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/hash"
+	commonlabels "github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/labels"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/elasticsearch/filesettings"
+	kblabel "github.com/elastic/cloud-on-k8s/v2/pkg/controller/kibana/label"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	KibanaConfigKey            = "kibana.json"
+	KibanaConfigHashAnnotation = "policy.k8s.elastic.co/kibana-config-hash"
+)
+
+func newKibanaConfigSecret(policy policyv1alpha1.StackConfigPolicy, kibana kibanav1.Kibana) (corev1.Secret, error) {
+	kibanaConfigHash := getKibanaConfigHash(policy.Spec.Kibana.Config)
+	var configDataJSONBytes []byte
+	var err error
+	if policy.Spec.Kibana.Config != nil {
+		configDataJSONBytes, err = policy.Spec.Kibana.Config.MarshalJSON()
+		if err != nil {
+			return corev1.Secret{}, err
+		}
+	}
+	kibanaConfigSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: kibana.Namespace,
+			Name:      GetPolicyConfigSecretName(kibana),
+			Labels: kblabel.NewLabels(types.NamespacedName{
+				Name:      kibana.Name,
+				Namespace: kibana.Namespace,
+			}),
+			Annotations: map[string]string{
+				KibanaConfigHashAnnotation: kibanaConfigHash,
```

Comment:
And this would be an empty annotation because the config hash would be `""`, right? Seems odd.

Reply:
I think even an empty config generates some hash.  I will double check

Reply:
This should be addressed with the change to only create the secret when config is present